docker-compose.yaml

```diff
@@ -30,6 +30,10 @@ services:
     container_name: sawtooth-settings-tp-local
     depends_on:
       - validator
+    environment:
```

Comment:
Why are you only adding the environment section to settings and validator?

Reply:
These are the components that do try to connect to resources in internet. If I remember it correctly, during cargo builds specifically. For other components they need internet access at docker image build time.

'args' tag in build will be consumed when docker image is getting baked.
But this is not persisted and passed to container.
For containers to get access to internet, they need to be passed with 'environment'.

Reply:
Will configuring docker client proxies in ~/.docker/config.json be enough as mentioned in https://docs.docker.com/network/proxy/ instead of using environment section, for Docker 17.07 or higher ?

Reply:
Yes